src/lib/LedgerApi.ts

```diff
@@ -328,39 +325,53 @@ class LedgerApi {
         }
 
         // prepare tx outside of request to avoid that an error would result in an endless loop in _callLedger
-        const senderPubKeyBytes = await LedgerApi.getPublicKey(keyPath);
-        const senderPubKey = new Nimiq.PublicKey(senderPubKeyBytes);
-        const senderAddress = senderPubKey.toAddress().toUserFriendlyAddress();
-        if (transaction.sender.replace(/ /g, '')
-            !== senderAddress.replace(/ /g, '')) {
-            LedgerApi._throwError(LedgerApi.ErrorType.REQUEST_ASSERTION_FAILED,
-                'Sender Address doesn\'t match this ledger account', request);
-        }
+        const [signerPubKeyBytes] = await Promise.all([
+            LedgerApi.getPublicKey(keyPath, walletId),
+            Nimiq.WasmHelper.doImportBrowser() // wasm needed for toAddress (also in BasicTransaction constructor)
+                .catch((e) => LedgerApi._throwError(LedgerApi.ErrorType.FAILED_LOADING_DEPENDENCIES, e)),
+        ]);
+        const signerPubKey = new Nimiq.PublicKey(signerPubKeyBytes);
+
+        const senderType = transaction.senderType !== undefined
+            ? transaction.senderType
+            : Nimiq.Account.Type.BASIC;
+
+        const recipientType = transaction.recipientType !== undefined
+            ? transaction.recipientType
+            : Nimiq.Account.Type.BASIC;
+
         const genesisConfig = Nimiq.GenesisConfig.CONFIGS[transaction.network];
         const networkId = genesisConfig.NETWORK_ID;
```

Comment:
If not using typescript, this will throw (`genesisConfig is undefined`) when `transaction.network` is not one of the possible strings or is left out. Is that on purpose?